packages/core/core/src/services/content-source-maps.ts

```diff
@@ -0,0 +1,201 @@
+import { vercelStegaCombine } from '@vercel/stega';
+import type { Core, Struct } from '@strapi/types';
+
+const ENCODABLE_TYPES = [
+  'string',
+  'text',
+  'richtext',
+  'uid',
+  'enumeration',
+  'email',
+  'date',
+  'datetime',
+  'time',
+  'timestamp',
+];
+
```

Comment:
shouldn't we encode all but relations, components & dzs? 
```suggestion
const ENCODABLE_TYPES = [
      "string",
      "text",
      "richtext",
      "integer",
      "blocks",
      "biginteger",
      "float",
      "decimal",
      "date",
      "time",
      "datetime",
      "timestamp",
      "boolean",
      "enumeration",
      "json",
      "media",
      "email",
      "password",
      "uid",
];
```

Reply:
Unfortunately stega encoding only works on strings, so I don't think we can attach metadata on numbers and booleans without changing the types of the response. Unless you have ideas :)

Blocks is intentionally missing because it will require some custom treatment that I'm keeping for later. But I'm probably still missing some. Thanks for your list, I'll check